projects/js-packages/publicize-components/src/components/add-connection-modal/confirmation-form/index.tsx

```diff
@@ -0,0 +1,265 @@
+import { Button, useGlobalNotices } from '@automattic/jetpack-components';
+import {
+	BaseControl,
+	FlexBlock,
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__experimentalHStack as HStack,
+} from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useMemo } from '@wordpress/element';
+import { __, _x } from '@wordpress/i18n';
+import { store as socialStore } from '../../../social-store';
+import { KeyringResult } from '../../../social-store/types';
+import { useSupportedServices } from '../use-supported-services';
+import styles from './style.module.scss';
+
+type ConfirmationFormProps = {
+	keyringResult: KeyringResult;
+	onComplete: VoidFunction;
+};
+
+type AccountOption = { label: string; value: string; profile_picture?: string };
+
+type AccountInfoProps = {
+	label: string;
+	profile_picture?: string;
+};
+
+/**
+ * Account info
+ *
+ * @param {AccountInfoProps} props - Component props
+ *
+ * @returns {import('react').ReactNode} Account info component
+ */
+function AccountInfo( { label, profile_picture }: AccountInfoProps ) {
+	return (
+		<div className={ styles[ 'account-info' ] }>
+			{ profile_picture ? (
+				<img className={ styles[ 'propfile-pic' ] } src={ profile_picture } alt={ label } />
+			) : null }
+			{ label }&nbsp;
+		</div>
+	);
+}
+
+/**
+ * Connection confirmation component
+ *
+ * @param {ConfirmationFormProps} props - Component props
+ *
+ * @returns {import('react').ReactNode} Connection confirmation component
+ */
+export function ConfirmationForm( { keyringResult, onComplete }: ConfirmationFormProps ) {
+	const supportedServices = useSupportedServices();
+	const { existingConnections, isCreatingConnection } = useSelect( select => {
+		const store = select( socialStore );
+
+		return {
+			existingConnections: store.getConnections(),
+			isCreatingConnection: store.isCreatingConnection(),
+		};
+	}, [] );
+
+	const { createErrorNotice } = useGlobalNotices();
+
+	const service = supportedServices.find(
+		supportedService => supportedService.ID === keyringResult.service
+	);
+	const isAlreadyConnected = useCallback(
+		( externalID: string ) => {
+			return existingConnections.some(
+				connection =>
+					connection.service_name === service?.ID && connection.external_id === externalID
+			);
+		},
+		[ existingConnections, service.ID ]
+	);
+
+	const { connected, not_connected } = useMemo( () => {
+		const options: Array< AccountOption > = [];
+
+		// Better safe than sorry
+		if ( service ) {
```

Comment:
Could we change this to:

```suggestion
		if ( ! service ) {
			return {}
		}
```

...and remove a layer of nesting?

Reply:
Done https://github.com/Automattic/jetpack/pull/37295/commits/35aa7c463d83a9874a5389472d681818f00d8e7f

Thanks